### PR TITLE
docs(icon): splitting icons into multiple pages

### PR DESCRIPTION
**Pull request by conor-darcy**
Too many icons on a single page takes too long to render (if they even render at all).
So I've split the icons into multiple pages.
Pages placed inside a 'docs' folder are generated & can be linked to.

Not perfect but it's an improvement.
Maybe other components could do with having sub-pages.
I will take another look at improving this in the near future, maybe an improved nav or adding them to the menu.

### CODE REVIEW

File: packages/genesys-spark-components/src/components/stable/gux-icon/docs/current.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<h1>gux-icon</h1>
+
+<nav>
+  <ul>
+    <li><a href="gux-icon-current.html">Current icons</a></li>
+    <li><a href="gux-icon-spark.html">Spark icons</a></li>
+    <li><a href="gux-icon-restricted.html">Restricted icons</a></li>
+    <li><a href="gux-icon-legacy.html">Legacy icons</a></li>
+    <li><a href="gux-icon-deprecated.html">Deprecated icons</a></li>
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-icon/docs/current.html:9]: I would add a link for the root/example.html page
```
<li><a href="gux-icon.html">Usage Guidelines</a></li>
```

File: packages/genesys-spark-components/src/components/stable/gux-icon/docs/current.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<h1>gux-icon</h1>
+
+<nav>
+  <ul>
+    <li><a href="gux-icon-current.html">Current icons</a></li>
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-icon/docs/current.html:5]: For completeness you should add a `aria-current="page"` attribute

**Comment by gavin-everett-genesys** [comment]
LGTM 👍  Would be nice if we could have an option like `Load more...` or some sort of infinite scrolling but there is probably accessibility problems associated with it and may be harder to find icons if the user does `CTRL + F` to look for icons.

**Comment by conor-darcy** [comment]
> LGTM 👍 Would be nice if we could have an option like `Load more...` or some sort of infinite scrolling but there is probably accessibility problems associated with it and may be harder to find icons if the user does `CTRL + F` to look for icons.

Yes Gavin, we can keep that in mind for the future.

**Review by gavin-everett-genesys — APPROVED** [review]

**Review by daragh-king-genesys — APPROVED** [review]

**Pull request merged**